### DISCUSSION TITLE
NodeJS support?

### DISCUSSION THREAD

**Discussion by luandro** [snakers4/silero-vad#416] [Ideas]
Would be great to have an example of how to use with NodeJS, maybe an npm package?

**Comment by csukuangfj** [comment]
Please have a look at

https://github.com/k2-fsa/sherpa-onnx/blob/master/nodejs-addon-examples/test_vad_microphone.js

The package name is
https://www.npmjs.com/package/sherpa-onnx-node


It supports using silero vad in NodeJS and provides an npm package, which perfectly matches your requirement.

Note it supports both silero vad version 4 and version 5.

**Comment by bargom** [comment]
Unfortunately for v5, it forces to use 16khz and 512 window size although the underlying model supports both 8khz and 16khz, it is a blocker for phone appliations (natively 8khz)

**Comment by Kamahl19** [comment]
@bargom have you figured a way to support 8khz?
